R/system.R

```diff
@@ -2451,28 +2451,44 @@ geocode_japan_prefecture <- function(df, prefecture_colname) {
-#' Currently, it can converts names into the short name format,
+#' Currently, with output_type ="name", it can converts names into the short name format,
 #' which has a name without the suffix such as "-to", "-ken".
+#' And with output_type = "code", it can converts names into the prefecture code.
 #'
 #' Example:
 #' > prefecturecode(c("東京都", "京都", "Kanagawa-ken", "Iwate", "あいち", "Kōchi", "gunma"), output_type="name")
 #' [1] "東京"   "京都"   "神奈川" "岩手"   "愛知"   "高知"    "群馬"
 
 prefecturecode <- function(prefecture, output_type="name") {
-  loadNamespace("stringr")
-  # TODO: support other output types.
-  # Clean up the input.
-  pref_normalized <- stringr::str_trim(tolower(prefecture))
-  # Remove trailing "tofuken". Do not remove "do" from "Hokkaido" and "to" from "Kyoto" (in Japanese).
-  pref_normalized <- dplyr::if_else(pref_normalized=="\u4EAC\u90FD", pref_normalized, gsub("[\u90FD\u5E9C\u770C]$", "", pref_normalized))
-  # Remove trailing "tofuken". Do not remove "do" from "Hokkaido" (in Roma-ji).
-  pref_normalized <- gsub("[_ \\.\\-](to|fu|hu|ken)$", "", pref_normalized)
-  # Convert "o" with macron to simple "o".
-  pref_normalized <- gsub("\u014D", "o", pref_normalized)
-  # Convert "gunma" to "gumma".
-  pref_normalized <- dplyr::if_else(pref_normalized=="gunma", "gumma" ,pref_normalized)
-  # Return the matching IDs.
-  return (as.character(jp_prefecture_name_id_map$id[match(pref_normalized, jp_prefecture_name_id_map$name)]))
+  if (output_type == "code") { # covert preferctur name to prefecture code
+    loadNamespace("zipangu")
+    prefecture = exploratory::prefecturecode(prefecture, output_type = "name")
+    targetDF <- data.frame(prefecture_normalized = prefecture)
+    answerDF <- zipangu::jpnprefs
+    # create prefecture_normalized column from a prefecture_kanji column and removes "都", "府", "県" from it.
+    answerDF <- answerDF %>% dplyr::mutate(prefecture_normalized = stringr::str_remove(prefecture_kanji, "[\u90FD\u5E9C\u770C]$"))
+    # Join data frames by prefecture_normalized so that it can get a corresponding prefecture code.
+    result <- targetDF %>% dplyr::left_join(answerDF, by = "prefecture_normalized")
+    result$jis_code
+  } else if (output_type == "name") {
+    loadNamespace("stringr")
+    # TODO: support other output types.
+    # Clean up the input.
+    pref_normalized <- stringr::str_trim(tolower(prefecture))
+    # Remove trailing "tofuken". Do not remove "do" from "Hokkaido" and "to" from "Kyoto" (in Japanese).
+    pref_normalized <- dplyr::if_else(pref_normalized=="\u4EAC\u90FD", pref_normalized, gsub("[\u90FD\u5E9C\u770C]$", "", pref_normalized))
+    # Remove trailing "tofuken". Do not remove "do" from "Hokkaido" (in Roma-ji).
+    pref_normalized <- gsub("[_ \\.\\-](to|fu|hu|ken)$", "", pref_normalized)
+    # Convert "o" with macron to simple "o".
+    pref_normalized <- gsub("\u014D", "o", pref_normalized)
+    # Convert "gunma" to "gumma".
```

Comment:
I think Gunma is the more officially used name.
https://www.pref.gunma.jp/04/c3610022.html

Reply:
Since both gunma and gumma give the correct prefecture code and name, I would rather leave it as is for now.

```r

> exploratory::prefecturecode(c("gunma", "gumma", "ぐんま", "群馬"), output_type = "code")
[1] "10" "10" "10" "10"
> exploratory::prefecturecode(c("gunma", "gumma", "ぐんま", "群馬"), output_type = "name")
[1] "群馬" "群馬" "群馬" "群馬"

```